### ISSUE TITLE
requiredIf doesn't work with return promise of boolean

### ISSUE THREAD

**Issue by jacobg** [vuelidate/vuelidate#605]
requiredIf doesn't seem to support the passed function returned a promise of a boolean. It looks like it considers the promise a truthy value, and thus the field is always required.

It would be great to support a promise.

**Comment by dobromir-hristov** [comment]
You are right, I just fixed it for V2. Will see if I can fix it for V0.x after the Holidays :)
